### PR TITLE
Add Liquity protocol

### PR DESCRIPTION
**Pull request by edmulraney**
Protocol: https://www.liquity.org/
Docs: https://docs.liquity.org/ (user focused) and https://github.com/liquity/dev (developer focused)

This connector adds support for interacting with Liquity protocol. Every function from Liquity has been proxied in the connector.

Users can:

### Perform Trove operations:
- Create, adjust and close a Trove

### Perform Stability Pool operations
- Deposit and withdraw LUSD from the Stability Pool
- Claim LQTY gains

### Perform LQTY staking operations:
- Stake and unstake LQTY
- Claim ETH and LUSD gains from Staking

### CODE REVIEW

**Review by thrilok209 — APPROVED** [review]

**Comment by thrilok209** [comment]
### Deployed connector:

**Connector Name**: "LIQUITY-A"
**Connector Address**: [0xEe619922B3a1334c798B0b756DD19077E34822C9](https://etherscan.io/address/0xEe619922B3a1334c798B0b756DD19077E34822C9)

**Comment by thrilok209** [comment]
Whitelisting Liquity connector transaction has been scheduled: https://etherscan.io/tx/0x44f5d4750cf07031bd738d7327898fcea3c993fbb90d73b0ffffe5f24da80167

It should be executable after Tue Jul 27 2021 16:43:20 GMT+0000.

File: contracts/mainnet/connectors/liquity/main.sol [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+     * @param setEthGainId Optional storage slot to store any ETH gains in
+     * @param setLqtyGainId Optional storage slot to store any LQTY gains in
+    */
+    function stabilityWithdraw(
+        uint amount,
+        uint getWithdrawId,
+        uint setWithdrawId,
+        uint setEthGainId,
+        uint setLqtyGainId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
```
Comment (KaymasJain) [inline contracts/mainnet/connectors/liquity/main.sol]:
```suggestion
    ) external payable returns (string memory _eventName, bytes memory _eventParam) {
```

File: contracts/mainnet/connectors/liquity/main.sol [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+    /**
+     * @dev Increase Trove collateral by sending Stability Pool ETH gain to user's Trove
+     * @notice Moves user's ETH gain from the Stability Pool into their Trove
+     * @param upperHint Address of the Trove near the upper bound of where the user's Trove should now sit in the ordered Trove list
+     * @param lowerHint Address of the Trove near the lower bound of where the user's Trove should now sit in the ordered Trove list
+    */
+    function stabilityMoveEthGainToTrove(
+        address upperHint,
+        address lowerHint
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
```
Comment (KaymasJain) [inline contracts/mainnet/connectors/liquity/main.sol]:
```suggestion
    ) external payable returns (string memory _eventName, bytes memory _eventParam) {
```

File: contracts/mainnet/connectors/liquity/main.sol [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+     * @param setEthGainId Optional storage slot to store any ETH gains
+     * @param setLusdGainId Optional storage slot to store any LUSD gains
+    */
+    function stake(
+        uint amount,
+        uint getStakeId,
+        uint setStakeId,
+        uint setEthGainId,
+        uint setLusdGainId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
```
Comment (KaymasJain) [inline contracts/mainnet/connectors/liquity/main.sol]:
```suggestion
    ) external payable returns (string memory _eventName, bytes memory _eventParam) {
```

File: contracts/mainnet/connectors/liquity/main.sol [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+     * @param setEthGainId Optional storage slot to store any ETH gains
+     * @param setLusdGainId Optional storage slot to store any LUSD gains
+    */
+    function unstake(
+        uint amount,
+        uint getUnstakeId,
+        uint setUnstakeId,
+        uint setEthGainId,
+        uint setLusdGainId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
```
Comment (KaymasJain) [inline contracts/mainnet/connectors/liquity/main.sol]:
```suggestion
    ) external payable returns (string memory _eventName, bytes memory _eventParam) {
```

File: contracts/mainnet/connectors/liquity/main.sol [hunk 5]
```diff
@@ -0,0 +1,458 @@
...
+    /**
+     * @dev Sends ETH and LUSD gains from Staking to user
+     * @notice Claim ETH and LUSD gains from Staking
+     * @param setEthGainId Optional storage slot to store the claimed ETH
+     * @param setLusdGainId Optional storage slot to store the claimed LUSD
+    */
+    function claimStakingGains(
+        uint setEthGainId,
+        uint setLusdGainId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
```
Comment (KaymasJain) [inline contracts/mainnet/connectors/liquity/main.sol]:
```suggestion
    ) external payable returns (string memory _eventName, bytes memory _eventParam) {
```

**Comment by thrilok209** [comment]
### Re-Deployed connector:

**Connector Name**: "LIQUITY-A"
**Connector Address**: [0xEe619922B3a1334c798B0b756DD19077E34822C9](https://etherscan.io/address/0xEe619922B3a1334c798B0b756DD19077E34822C9)

**Comment by thrilok209** [comment]
Whitelisting Liquity connector transaction has been re-scheduled: https://etherscan.io/tx/0x1bef4af8d88823f76546f3c1cb3094126ce7ecc0811da1a9c60283b8941f7472

It should be executable after Thu Aug 05 2021 12:53:10 GMT+0000.

**Comment by thrilok209** [comment]
Liquitiy connector whitelisted: https://etherscan.io/tx/0x4976e2166d638be68fb1e4f9c4cc6cdf3b72c2ce17e4baad23766344a43dda30

**Pull request merged**